src/txn/txn_recover.c

```diff
@@ -1192,7 +1192,7 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[], bool disagg)
      * 3. We are not using disaggregated storage or precise checkpoint
      */
     if (hs_exists_local && !F_ISSET(conn, WT_CONN_READONLY) && !disagg &&
-      !F_ISSET(conn, WT_CONN_PRECISE_CHECKPOINT)) {
+      !F_ISSET_ATOMIC_32(conn, WT_CONN_PRECISE_CHECKPOINT)) {
```

Comment:
changed this check to `F_ISSET_ATOMIC_32` since this [commit](https://github.com/wiredtiger/wiredtiger/commit/ea31381eab3a6e4c6b33246344ec35d472847768) made all non-read-only connection flags atomic which caused this PR to fail

Reply:
made this flag non-atomic